src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -424,12 +427,31 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
       placement.inlineSize,
     ]);
 
+    useEffect(() => {
+      // The disabledMotionGlobalClass class is initially applied to the root element.
+      // It disables all animations until the first user interaction,
+      // which prevents dynamic elements intended as "progressive enhancement"
+      // from animating prematurely. This approach ensures animations
+      // are triggered only by user interactions.
+      function enableAnimations() {
+        rootRef.current?.classList.remove(disabledAnimationsGlobalClass);
+        document.removeEventListener('click', enableAnimations);
```

Comment:
1. Yes, it's fine, since the css class removal happens imperatively, so we don't need to wait the next react update cycle. Tested under multiple conditions manually.
2. Keyboard events trigger click events when called on CTAs (like trigger buttons)
3. Transitions might happen only if animations are enabled, which is not a case here

Reply:
1. Yes, it seems to work fine for the current setup, but I see some future proof issues. At the very least I suggest to switch from capturing to the bubble mode
2. Ok, got it
3. Imagine this use-case. 1) separate app layout instances 2) you interacted with the first app layout, enabled animations 3) you switched on a different page in SPA, no page reload 3) the app layout on the next page is suffering from the original problem

Reply:
1. Switching to capture mode sounds reasonable. Addressed
2. -
3. Fair point. Technically I can take into account the current page visibility and add/remove motion disabled class.  But now I have doubts if we even need this animation disabling. I would consider an option to talk with console home to simplify their `navigationOpen` definition

Reply:
Summary from the sync today.

3. For this one I still recommend keeping animations only active some some short duration. Mathematically, there is a bigger part of the page lifetime when animations are not needed, then when needed.

it could be something like this

```js
document.addEventListener('click', () => {
 enableAnimations();
 setTimeout(() => disableAnimations(), 250)
}, true);
```

4. For the iframe use-case. We probably should walk up window parents to capture all possible interactions (or maybe there is some convenient API in browsers for that)

```js
while(parent = window.parent) {
   parent.document.addEventListener('click')
}
```

Reply:
Here's what I tried: enabling and disabling animations right after user interaction led to a race condition, making it difficult to determine the correct times for toggling animations during route navigation. Moreover, the following approach doesn't work:

```js
while (parent = window.parent) {
   parent.document.addEventListener('click')
}
```

It requires the use of the postMessage API, which could lead to significant maintenance challenges afterwards.

So, I introduced a different approach. I checked `defaultActive` in drawers and the first user interaction with the navigation component. From my perspective, it simplifies implementation and maintenance